### ISSUE TITLE
import/no-duplicates: False positives with Meteor imports

### ISSUE THREAD

**Issue by alisnic** [oxc-project/oxc#13151]
```
  × eslint-plugin-import(no-duplicates): Modules should not be imported multiple times in the same file
   ╭─[client/linkedin-client.ts:3:38]
 1 │ import { OAuth } from 'meteor/oauth';
   ·                       ──────────────
 2 │ import { Random } from 'meteor/random';
   ·                        ───────────────
 3 │ import { ServiceConfiguration } from 'meteor/service-configuration';
```

I tried to write a failing spec for it in the codebase and vibe-code a solution, but the the tests pass when I do that 🤔

```
let pass = vec![
  //
  (
      r"import { Accounts } from 'meteor/accounts-base'; import { Meteor } from 'meteor/meteor'",
      None,
  ),
]
```

**Comment by camc314** [comment]
Hi, i tried to repro this issue, and was unable to?

could you share:
 -  your oxlint version
 - whether you're using the LSP or CLI
 - the full contents of the file that can reproduce the issue
 - when this broke? did it break after an new oxlint version, or some other change? or when using oxlint for the first time

So that I can narrow down the cause

**Comment by alisnic** [comment]
Here is a repo that reproduces the issue https://github.com/alisnic/oxlint-duplicate-imports-repro/tree/master. I think the problem is in how meteor exports its package types. It generates a huge packages.d.ts file which exports all module types.

I assume oxlint considers modules with type definitions originating from the same file as duplicates

**Comment by Afsoon** [comment]
This bug can be closed as fixed. Since version 1.37, oxlint doesn't lint this as an error. I can't link exactly what PR fixed. 

<img width="1060" height="783" alt="Image" src="https://github.com/user-attachments/assets/bb74d454-c253-4f64-9ca5-e10087f95601" />

**Comment by camc314** [comment]
Thanks Afsoon!

I suspect something got updated in oxc_resolver to fix this issue

@alisnic please reopen if you can still reproduce this!

**Comment by alisnic** [comment]
It seems that this regressed in 1.58

```
  × eslint-plugin-import(no-duplicates): Modules should not be imported multiple times in the same file
   ╭─[imports/api/public/index.js:4:24]
 3 │ import { Meteor } from 'meteor/meteor';
 4 │ import { Picker } from 'meteor/quave:picker';
   ·                        ──────────┬──────────
   ·                                  ╰── It is first imported here
 5 │ import { Random } from 'meteor/random';
 6 │ import { WebApp } from 'meteor/webapp';
   ·                        ───────────────
 7 │ import { traceIdContext } from '/imports/api/services/asyncContext';
   ╰────
  help: Merge these imports into a single import statement
```

I don't have access to reopen
